spec/text-editor-spec.coffee

```diff
@@ -4018,3 +4018,71 @@ describe "TextEditor", ->
       editor.setPlaceholderText('OK')
       expect(handler).toHaveBeenCalledWith 'OK'
       expect(editor.getPlaceholderText()).toBe 'OK'
+
+  describe 'Gutters', ->
+    describe 'Add/Remove Gutters', ->
+      it 'can add a right gutter', ->
+        options = priority: 1
+        gutter = editor.addRightGutter 'test-gutter', options
+        expect(editor.getRightGutters().length).toBe 1
+        expect(editor.getRightGutters()[0]).toBe gutter
+
+      it 'can add a left gutter', ->
+        gutter = editor.addLeftGutter 'test-gutter'
+        expect(editor.getLeftGutters().length).toBe 1
+        expect(editor.getLeftGutters()[0]).toBe gutter
+
+      it 'will notify listeners when a gutter is removed', ->
+        mockListener = onGutterRemoved: null
+        editor.onDidRemoveGutter (gutter) ->
+          mockListener.onGutterRemoved(gutter)
+        spyOn mockListener, 'onGutterRemoved'
+
+        gutter = editor.addLeftGutter 'test-gutter'
+        gutter.destroy()
+        expect(mockListener.onGutterRemoved).toHaveBeenCalledWith(gutter)
+
+    describe 'Show/Hide Gutters', ->
+      mockListener =
+      onGutterHidden: null
+      onGutterShown: null
+
+      beforeEach ->
+        editor.onDidShowGutter (gutter) ->
+          mockListener.onGutterShown(gutter)
+        editor.onDidHideGutter (gutter) ->
+          mockListener.onGutterHidden(gutter)
+        spyOn mockListener, 'onGutterShown'
+        spyOn mockListener, 'onGutterHidden'
+
+      it 'will notify listeners when a gutter is shown', ->
+        options = visibleInitially: false
+        gutter = editor.addRightGutter 'test-gutter', options
+        gutter.show()
+        expect(mockListener.onGutterShown).toHaveBeenCalledWith(gutter)
+
+      it 'will notify listeners when a gutter is hidden', ->
+        options = visibleInitially: true
+        gutter = editor.addRightGutter 'test-gutter', options
+        gutter.hide()
+        expect(mockListener.onGutterHidden).toHaveBeenCalledWith(gutter)
+
+    describe 'Default Gutter', ->
+      it 'has a non-nil default gutter', ->
+        expect(editor.getDefaultGutter()).not.toBeNull()
+
+    describe 'Decorating Gutters', ->
+      it 'adds a decoration when one of its gutters has decorateMarker called on it', ->
```

Comment:
>  It might be a good idea to force the height of a gutter item to match
the line height.

I have already put things in the gutter that don't match the line height.
This is not problem if there is an easy way to override with a percentage.

On Thu, Feb 5, 2015 at 4:15 PM, Nathan Sobo <notifications@github.com>
wrote:

> In spec/text-editor-spec.coffee
> <https://github.com/atom/atom/pull/5217#discussion_r24213069>:
>
> > +        gutter = editor.addRightGutter 'test-gutter', options
> > +        gutter.show()
> > +        expect(mockListener.onGutterShown).toHaveBeenCalledWith(gutter)
> > +
> > +      it 'will notify listeners when a gutter is hidden', ->
> > +        options = visibleInitially: true
> > +        gutter = editor.addRightGutter 'test-gutter', options
> > +        gutter.hide()
> > +        expect(mockListener.onGutterHidden).toHaveBeenCalledWith(gutter)
> > +
> > +    describe 'Default Gutter', ->
> > +      it 'has a non-nil default gutter', ->
> > +        expect(editor.getDefaultGutter()).not.toBeNull()
> > +
> > +    describe 'Decorating Gutters', ->
> > +      it 'adds a decoration when one of its gutters has decorateMarker called on it', ->
>
>  I was thinking it makes sense to have the height of a row defined by the
> height of the line number gutter in that row... actually thinking about it
> now we should probably chat a bit how you were envisioning a custom gutter
> "item" being positioned.
>
> Yeah, that makes sense. It might be a good idea to force the height of a
> gutter item to match the line height... not sure. I was thinking we'd just
> absolutely position the item in the appropriate location.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/atom/atom/pull/5217/files#r24213069>.
>